bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/discovery/SiemenesHvacDiscoveryParticipant.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.siemenshvac.internal.discovery;
+
+import static org.openhab.core.thing.Thing.PROPERTY_SERIAL_NUMBER;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.openhab.binding.siemenshvac.internal.constants.SiemensHvacBindingConstants;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.upnp.UpnpDiscoveryParticipant;
+import org.openhab.core.config.discovery.upnp.internal.UpnpDiscoveryService;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * The {@link SiemensHvacDiscoveryParticipant} is responsible for discovering new and
+ * removed siemensHvac bridges. It uses the central {@link UpnpDiscoveryService}.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = UpnpDiscoveryParticipant.class, configurationPid = "discovery.siemenshvac")
+public class SiemenesHvacDiscoveryParticipant implements UpnpDiscoveryParticipant {
+
+    @Activate
+    public void activate(@Nullable Map<String, Object> configProperties) {
+    }
+
+    @Modified
+    public void modified(@Nullable Map<String, Object> configProperties) {
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Set.of(SiemensHvacBindingConstants.THING_TYPE_BRIDGE);
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(RemoteDevice device) {
+        ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>();
+            String ipAddress = device.getDetails().getPresentationURI().getHost();
+            properties.put(SiemensHvacBindingConstants.IP_ADDRESS, ipAddress);
```

Comment:
I use it to create the bridge thing UID.
This can help differenciate bridge if there is more than one bridge on same network (not very common !).

Reply:
I don't understand the correlation between providing IP address as a property and using to when generating a bridge UID? Also the user can freely define the bridge UID (i.e. when manually creating a bridge, including file-based configuration), so you cannot rely on information being embedded into the UID.

Related to https://github.com/openhab/openhab-addons/pull/14263/files#r1598335493

Reply:
There is not direct correlation.
The information is just put in the properties to give the end user some overview of what bridge is handling what physical device.
If information is not in the UID (because user defined manually the id), user can still find / verify the physical device looking at the properties

Reply:
In that case the property should be provided by the handler, not by discovery. This way you will always see the same property, no matter if you created the Thing from discovery or manually.

Reply:
Jacob,

I propose finally to remove this property.
It is not really use, and is redudant with the URL parameter that is already on the OZW Gateway.

Reply:
Can you check `getIPBridgeThingUID` in `SiemensHvacHandlerFactory`, it seems to depend on IP address being configured?

Reply:
I completly remove this function

In fact, it is only call from createThing, and thingUid is always already initialized that we come from autodiscover or manual creation.

So we never go through the second part of the code getting the IP from configuration.
And the function is totally useless.


```
   private ThingUID getIPBridgeThingUID(ThingTypeUID thingTypeUID, @Nullable ThingUID thingUID,
            Configuration configuration) {
        if (thingUID != null) {
            return thingUID;
        }
        String ipAddress = (String) configuration.get(SiemensHvacBindingConstants.IP_ADDRESS);
        return new ThingUID(thingTypeUID, ipAddress);
    }
```

Reply:
fix in next commit